File: android/app/src/main/java/com/yourapp/UgoiraView.java
package com.yourapp;

import android.content.Context;
import android.graphics.Bitmap;
import android.graphics.BitmapFactory;
import android.graphics.Canvas;
import android.util.AttributeSet;
import android.view.MotionEvent;
import android.view.ViewGroup;
import com.facebook.react.bridge.ReactMethod;
import com.facebook.react.uimanager.SimpleViewManager;
import com.facebook.react.uimanager.ThemedReactContext;
import com.facebook.react.uimanager.annotations.ReactProp;

public class UgoiraView extends ViewGroup {
    private Bitmap ugoiraImage;

    public UgoiraView(Context context) {
        super(context);
    }

    public UgoiraView(Context context, AttributeSet attrs) {
        super(context, attrs);
    }

    public UgoiraView(Context context, AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
    }

    public void loadImage(String imagePath) {
        if (ugoiraImage != null) {
            ugoiraImage.recycle();
        }
        ugoiraImage = BitmapFactory.decodeFile(imagePath);
        invalidate();
    }

    @Override
    protected void onDraw(Canvas canvas) {
        super.onDraw(canvas);
        if (ugoiraImage != null) {
            canvas.drawBitmap(ugoiraImage, 0, 0, null);
        }
    }

    @Override
    public boolean onTouchEvent(MotionEvent event) {
        // Handle touch events
        return true;
    }

    @Override
    protected LayoutParams generateDefaultLayoutParams() {
        return new LayoutParams(LayoutParams.MATCH_PARENT, LayoutParams.MATCH_PARENT);
    }

    @Override
    protected LayoutParams generateLayoutParams(AttributeSet attrs) {
        return new LayoutParams(getContext(), attrs);
    }

    @Override
    protected void onLayout(boolean changed, int left, int top, int right, int bottom) {
        layout(0, 0, right - left, bottom - top);
    }
}

class UgoiraViewManager extends SimpleViewManager<UgoiraView> {
    public static final String REACT_CLASS = "UgoiraView";

    @Override
    public String getName() {
        return REACT_CLASS;
    }

    @Override
    protected UgoiraView createViewInstance(ThemedReactContext reactContext) {
        return new UgoiraView(reactContext);
    }

    @ReactMethod
    public void loadImage(String imagePath, int viewId) {
        UgoiraView ugoiraView = (UgoiraView) getViewById(viewId);
        if (ugoiraView != null) {
            ugoiraView.loadImage(imagePath);
        }
    }
}

File: android/app/src/main/kotlin/com/example/yourapp/UgoiraView.kt
package com.example.yourapp

import android.content.Context
import android.graphics.Bitmap
import android.graphics.BitmapFactory
import android.graphics.Canvas
import android.util.AttributeSet
import android.view.View
import com.facebook.react.bridge.ReadableArray
import java.util.Timer
import java.util.TimerTask

class UgoiraView(context: Context, attrs: AttributeSet? = null) : View(context, attrs) {
    private var images: List<Bitmap> = emptyList()
    private var currentIndex: Int = 0
    private var timer: Timer? = null
    private var isAnimating: Boolean = false
    var scaleType: ScaleType = ScaleType.CENTER_CROP

    enum class ScaleType {
        CENTER_CROP, FIT_CENTER
    }

    fun setImages(imageArray: ReadableArray) {
        images = List(imageArray.size()) { index ->
            val imagePath = imageArray.getString(index)
            try {
                BitmapFactory.decodeFile(imagePath) ?: Bitmap.createBitmap(1, 1, Bitmap.Config.ARGB_8888)
            } catch (e: Exception) {
                Bitmap.createBitmap(1, 1, Bitmap.Config.ARGB_8888)
            }
        }
        currentIndex = 0
        invalidate()
    }

    fun setScaleType(newScaleType: ScaleType) {
        scaleType = newScaleType
        invalidate()
    }

    fun pauseAnimation() {
        isAnimating = false
        timer?.cancel()
    }

    fun resumeAnimation() {
        if (!isAnimating && images.isNotEmpty()) {
            isAnimating = true
            startAnimation()
        }
    }

    private fun startAnimation() {
        timer = Timer()
        timer?.scheduleAtFixedRate(object : TimerTask() {
            override fun run() {
                if (isAnimating && images.isNotEmpty()) {
                    currentIndex = (currentIndex + 1) % images.size
                    postInvalidate()
                }
            }
        }, 0, 100)
    }

    override fun onDraw(canvas: Canvas) {
        super.onDraw(canvas)
        if (images.isNotEmpty()) {
            val bitmap = images[currentIndex]
            val scaledBitmap = Bitmap.createScaledBitmap(bitmap, measuredWidth, measuredHeight, true)
            when (scaleType) {
                ScaleType.CENTER_CROP -> {
                    val scale = Math.max(measuredWidth.toFloat() / scaledBitmap.width, measuredHeight.toFloat() / scaledBitmap.height)
                    val scaledWidth = (scaledBitmap.width * scale).toInt()
                    val scaledHeight = (scaledBitmap.height * scale).toInt()
                    val left = (measuredWidth - scaledWidth) / 2
                    val top = (measuredHeight - scaledHeight) / 2
                    canvas.drawBitmap(Bitmap.createScaledBitmap(scaledBitmap, scaledWidth, scaledHeight, true), left.toFloat(), top.toFloat(), null)
                }
                ScaleType.FIT_CENTER -> {
                    val scale = Math.min(measuredWidth.toFloat() / scaledBitmap.width, measuredHeight.toFloat() / scaledBitmap.height)
                    val scaledWidth = (scaledBitmap.width * scale).toInt()
                    val scaledHeight = (scaledBitmap.height * scale).toInt()
                    val left = (measuredWidth - scaledWidth) / 2
                    val top = (measuredHeight - scaledHeight) / 2
                    canvas.drawBitmap(Bitmap.createScaledBitmap(scaledBitmap, scaledWidth, scaledHeight, true), left.toFloat(), top.toFloat(), null)
                }
            }
        }
    }
}